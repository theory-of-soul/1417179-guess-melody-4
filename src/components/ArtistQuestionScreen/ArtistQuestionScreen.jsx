import React from "react";
import PropTypes from "prop-types";

const ArtistQuestionScreen = (props) => {
  const {
    question,
    handleAnswer
  } = props;
  return (
    <section className="game game--artist">
      <header className="game__header">
        <a className="game__back" href="#">
          <span className="visually-hidden">Сыграть ещё раз</span>
          <img className="game__logo" src="img/melody-logo-ginger.png" alt="Угадай мелодию" />
        </a>

        {/*  <svg xmlns="http://www.w3.org/2000/svg" className="timer" viewBox="0 0 780 780">*/}
        {/*  <circle className="timer__line" cx="390" cy="390" r="370"*/}
        {/*    style="filter: url(#blur); transform: rotate(-90deg) scaleY(-1); transform-origin: center"/>*/}
        {/*  </svg>*/}

        <div className="game__mistakes">
          <div className="wrong"></div>
          <div className="wrong"></div>
          <div className="wrong"></div>
        </div>
      </header>

      <section className="game__screen">
        <h2 className="game__title">Кто исполняет эту песню?</h2>
        <div className="game__track">
          <div className="track">
            <button className="track__button track__button--play" type="button"></button>
            <div className="track__status">
              <audio></audio>
            </div>
          </div>
        </div>

        <form className="game__artist">
          {
            question.answers.map((answer, index) => {
              return (
                <div className="artist" key={`${answer.name}`}>
                  <input
                    className="artist__input visually-hidden"
                    type="radio"
                    name="answer"
                    value={`artist-${index}`}
                    id={`answer-${index}`}
                  />
                  <label
                    className="artist__name"
                    htmlFor={`answer-${index}`}
                    onClick={() => {
                      handleAnswer(question, answer);
                    }}
                  >
                    <img className="artist__picture" src={answer.pic} alt={answer.name} />
                    {answer.name}
                  </label>
                </div>
              );
            })
          }
        </form>
      </section>
    </section>
  );
};

ArtistQuestionScreen.propTypes = {
  question: PropTypes.shape({
    type: PropTypes.string.isRequired,
    rightAnswer: PropTypes.string.isRequired,
    answers: PropTypes.arrayOf(
        PropTypes.shape({
          pic: PropTypes.string.isRequired,
          name: PropTypes.string.isRequired,
        })
    ).isRequired
  }),
  handleAnswer: PropTypes.func.isRequired
};

export default ArtistQuestionScreen;
