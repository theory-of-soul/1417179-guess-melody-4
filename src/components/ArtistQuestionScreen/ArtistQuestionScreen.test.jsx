import React from "react";
import renderer from 'react-test-renderer';
import ArtistQuestionScreen from "./ArtistQuestionScreen";

const question = {
  type: `artist`,
  rightAnswer: `Пелагея`,
  answers: [{
    pic: `http://placehold.it/134x134`,
    name: `Пелагея`
  }, {
    pic: `http://placehold.it/134x134`,
    name: `Краснознаменная дивизия имени моей бабушки`
  }, {
    pic: `http://placehold.it/134x134`,
    name: `Lorde`
  }]
};

describe(`ArtistQuestionScreen snapshot tests`, () => {
  it(`ArtistQuestionScreen should render 3 singers`, () => {
    const component = renderer
      .create(<ArtistQuestionScreen question={question} handleAnswer={() => {}} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
