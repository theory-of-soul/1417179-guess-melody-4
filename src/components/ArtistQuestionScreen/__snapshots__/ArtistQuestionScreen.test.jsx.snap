// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen snapshot tests ArtistQuestionScreen should render 3 singers 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-0"
          name="answer"
          type="radio"
          value="artist-0"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
          onClick={[Function]}
        >
          <img
            alt="Пелагея"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Пелагея
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          type="radio"
          value="artist-1"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
          onClick={[Function]}
        >
          <img
            alt="Краснознаменная дивизия имени моей бабушки"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Краснознаменная дивизия имени моей бабушки
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-2"
          name="answer"
          type="radio"
          value="artist-2"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
          onClick={[Function]}
        >
          <img
            alt="Lorde"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Lorde
        </label>
      </div>
    </form>
  </section>
</section>
`;
